crates/node/service/src/service/validator.rs

```diff
@@ -179,20 +181,13 @@ pub trait ValidatorNodeService {
             },
         );
 
-        let rollup_rpc = RollupRpc::new(engine_query_sender, l1_watcher_queries_sender);
-
         // The RPC Server should go last to let other actors register their rpc modules.
-        let rpc = if let Some(mut rpc) = self.rpc() {
-            if let Some(p2p_module) = p2p_module {
-                rpc = rpc.merge(p2p_module.into_rpc()).map_err(Self::Error::from)?;
-            }
-
-            rpc = rpc.merge(rollup_rpc.into_rpc()).map_err(Self::Error::from)?;
-            let handle = rpc.start().await?;
-            Some(RpcActor::new(handle, cancellation.clone()))
-        } else {
-            None
-        };
+        let mut launcher = self.rpc();
+        launcher = launcher.merge(p2p_module.map(|r| r.into_rpc())).map_err(Self::Error::from)?;
+        let rollup_rpc = RollupRpc::new(engine_query_sender, l1_watcher_queries_sender);
+        launcher = launcher.merge(Some(rollup_rpc.into_rpc())).map_err(Self::Error::from)?;
+        let handle = launcher.launch().await?;
+        let rpc = Some(RpcActor::new(handle, cancellation.clone()));
```

Comment:
RPC is no longer optional?

Reply:
It previously wasn't, but I've went ahead and made it optional via cli in this PR